### PR TITLE
implementation of formatting for element attributes

### PR DESCRIPTION
**Pull request by twoseascharlie**
proposed server side fix for:
redhat-developer/vscode-xml#633

related client side documentation update:
redhat-developer/vscode-xml#636

Creating this as a draft PR to get early feedback on this approach.
If the change looks like a good approach, automated tests
and documentation updates will follow in a separate commit

@angelozerr Can you take a look and let me know what you think please?

Here's a screenshot of the changes in the outline using the below json setting:
newOutline

```json
        "pattern": "build*.xml",
            "expressions" :[
                {
                    // keep the value of the attribute "file" on the same line
                    // as the "import" element, and only show the attribute value
                    "xpath": "//import/@file"
                },
                {
                    // keep the value of the attribute "name" on the same line
                    // as the "target" element, and only show the attribute value
                    "xpath": "//target/@name",
                    "primaryElementAttr" : true,
                    "valueOnly": true
                },
                // show "unless" and "depends" as nested attributes for
                // "target" elements
                {
                "xpath": "//target/@unless"
                },
                {
                "xpath": "//target/@depends"
                },
                {
                    // keep the value of the attribute "name" on the same line
                    // as the "property" element, and only show the attribute value
                    "xpath": "//property/@name",
                    "primaryElementAttr" : true,
                    "valueOnly": true
                },
                {
                    // keep the value of the attribute "file" on the same line
                    // as the "property" element, and show the attribute name
                    // along with the value to distinguish it from the more common
                    // "name" attribute
                    "xpath": "//property/@file",
                    "primaryElementAttr" : true,
                    "valueOnly": false
                }
```

### CODE REVIEW

**Comment by angelozerr** [comment]
It's look really promising, thanks so much for your contribution.

No rush, but just for your information I think we will do soon a new release, so if you have time we could embed your contribution. We will see...

@fbricon could you give us your opinion about the new settings name `primaryElementAttr`, and `valueOnly `please.

@twoseascharlie it should be nice if you could:

 * write several tests like https://github.com/eclipse/lemminx/blob/29530bc21168c8ca0836579a83643918b2be425c/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLDocumentSymbolsTest.java#L377
 * write several tests like https://github.com/eclipse/lemminx/blob/29530bc21168c8ca0836579a83643918b2be425c/org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/XMLSymbolInformationsTest.java#L345
 * update the doc for XML symbol filter and attribute at https://github.com/redhat-developer/vscode-xml/blob/master/docs/Symbols.md#attributes-sample

Otherwise I will review your PR ASAP.

**Comment by fbricon** [comment]
Can we have more descriptive properties like:
```
inlineAttribute=true
showAttributeValue=true
```

**Comment by fbricon** [comment]
showAttributeName works for me

**Comment by twoseascharlie** [comment]
Didn't have as much time to work on this today as I had hoped. I will be offline over the weekend, but I should have the PR ready monday or tuesday.

**Comment by angelozerr** [comment]
> Didn't have as much time to work on this today as I had hoped. I will be offline over the weekend, but I should have the PR ready monday or tuesday.

No rush @twoseascharlie ! It's not sure that we do a release soon, so take your time.

**Comment by twoseascharlie** [comment]
I've updated my commits with my gpg signature, sorry for the noise

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolExpressionFilter.java [hunk 1]
```diff
@@ -45,6 +49,55 @@ public void setXpath(String xpath) {
...
+	/**
+	 * Returns true if the filter which matches an element attribute node should show
+	 * the attribute name or not, the attribute value will always be shown, and false
+	 * otherwise.
+	 * 
+	 * @return true if the filter which matches an element attribute node should show
+	 *         the attribute name or not, the attribute value will always be shown, and false
+	 *         otherwise.
+	 */
+	public boolean shouldShowAttributeName() {
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolExpressionFilter.java:85]: Rename to `isShowAttributeName`

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolExpressionFilter.java [hunk 2]
```diff
@@ -45,6 +49,55 @@ public void setXpath(String xpath) {
...
 
+	/**
+	 * Returns true if the filter which matches an element attribute node must consider it
+	 * an inline attribute and show it on the same line as the owning element and false otherwise.
+	 * 
+	 * @return true if the filter which matches an element attribute node must consider it
+	 *         an inline attribute and show it on the same line as the owning element and
+	 *         false otherwise.
+	 */
+	public boolean isinlineAttribute() {
```
Comment (angelozerr) [inline org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolExpressionFilter.java:60]: Rename to `isInlineAttribute`

Comment (twoseascharlie) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolExpressionFilter.java]: nice catch, updated

File: org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolExpressionFilter.java (hunk 1 above)
Comment (twoseascharlie) [reply org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/settings/XMLSymbolExpressionFilter.java]: updated

**Comment by angelozerr** [comment]
Could you squash your commits in one commit please

**Comment by fbricon** [comment]
@angelozerr it's fine we can squash and merge directly from GitHub

**Comment by angelozerr** [comment]
@twoseascharlie could you add a **Signed-off-by: ???** in your commit please

**Comment by angelozerr** [comment]
Once **Signed-off-by:**  is fixed, I'm OK to merge this great PR.

**Comment by twoseascharlie** [comment]
@angelozerr I've added the signed off by to my commit. Thanks for taking a look at this one!

**Comment by angelozerr** [comment]
Thanks so much @twoseascharlie for your great PR!

Dont hesitate to contribute to lemminx with other great ideas!

**Pull request merged**